example/server/integrations.py

```diff
@@ -388,7 +390,69 @@ def process_sep6_request(self, params: Dict) -> Dict:
 
 
 class MyCustomerIntegration(CustomerIntegration):
-    def put(self, params: Dict):
+    def get(self, params: Dict) -> Dict:
+        query_params = {}
+        for attr in ["id", "memo", "memo_type", "account"]:
+            if attr in params:
+                query_params[attr] = params.get(attr)
+        account = PolarisStellarAccount.objects.filter(**query_params).first()
+        if "id" in query_params and not account:
+            # client believes the customer already exists but it doesn't,
+            # at least not with the same ID, memo, account values.
+            raise ValueError(
+                _("customer not found using: %s") % list(query_params.keys())
+            )
+        elif not account:
+            response = {
+                "status": "NEEDS_INFO",
+                "fields": {
+                    "first_name": {"description": "first name of the customer"},
+                    "last_name": {"description": "last name of the customer"},
+                    "email_address": {"description": "email address of the customer"},
+                },
+            }
+            if params.get("type") in ["sep6-deposit", "sep31-sender"]:
+                return response
+            elif params.get("type") in [None, "sep6-withdraw", "sep31-receiver"]:
+                response["fields"].update(
+                    {
+                        "bank_account_number": {
```

Comment:
All this logic feels kind of messy, it'd be nice to have a simple mapping of what fields are needed by what types, and then the logic can just iterate through that rather than put the needed fields in all the if statements